src/main/java/seedu/address/model/Wallet.java

```diff
@@ -123,6 +129,58 @@ public void setExpense(Expense target, Expense editedExpense) {
     public void removeExpense(Expense key) {
         expenses.remove(key);
     }
+
+    public Amount getTotalExpenditureInMonth(Date date) {
+        requireNonNull(date);
+        TransactionList<Expense> filteredExpenseList = expenses.getTransactionsInMonth(date);
+
+        return filteredExpenseList.getTotal();
+    }
+
+    // =========== Budget-related Operations =============================================================
+
+    /**
+     * Replaces the budget in the Wallet with {@code budget}. If a budget was previously set, it overrides that
+     * budget and sets this new value.
+     */
+    public void setBudget(Budget budget) {
+        requireNonNull(budget);
+        budgetList.setBudget(budget);
+    }
+
+    /**
+     * Replaces the default budget in the Wallet with {@code budget}.
+     */
+    public void setDefaultBudget(Budget budget) {
+        requireNonNull(budget);
+        budgetList.setDefaultBudget(budget);
+    }
+
+    /**
+     * Checks if the budget has been exceeded on a given {@code date}.
+     */
+    public boolean hasExceededBudget(Date date) {
```

Comment:
This is a bit confusing because it suggests that it would only check expenses up till a certain date, but it checks expenses for the whole month the date is in. Maybe using month would be better